### DISCUSSION TITLE
V5: error handling for actions

### DISCUSSION THREAD

**Discussion by christophe-g** [statelyai/xstate#4505] [Q&A] [answered]
I am not sure if this need to be reported as a bug, but I am having issues with how to handle unhandled  errors in actions (works well with actor logic like `fromPromise`, ...).

I am expecting 2 things from the  following scenario: 
1. the `unhandled_sync_error_on_state_entry` error is caught on `actorRef.subscribe({error: ...})`, and
2. the latest snapshot is in an `error` status.

**1.** is satisfied in my local tests, but not **2.** (status is `active`)

```js
const machine = createMachine({
      initial: 'pending',
      states: {
        pending: {
          entry: () => {
            throw new Error('unhandled_sync_error_on_state_entry');
          }
        }
      }
    });

    const actorRef = createActor(machine);
    actorRef.subscribe({
      error: (error) => {
       // the error 'unhandled_sync_error_on_state_entry' sync error is caught here
      }
    });
    actorRef.start();
    // actorRef.getSnapshot().status is error

```

I tried to add this test in the error test suite as I think this should be the expected behavior, but it fails (just note that I am new to `xstate` and might miss smth very stupid, so bear with me !)

```js
  it(`unhandled sync errors thrown on state entry should crash the parent`, (done) => {
    const machine = createMachine({
      initial: 'pending',
      states: {
        pending: {
          entry: () => {
            throw new Error('unhandled_sync_error_on_state_entry');
          }
        }
      }
    });

    const actorRef = createActor(machine);
    actorRef.subscribe({
      error: (error) => {
        expect((error as Error).message).toBe('unhandled_sync_error_on_state_entry');
        // error is caught here, we throw for upstream (global)
        throw new Error('unhandled_sync_error_on_state_entry_global');
      }
    });
    actorRef.start();

    expect(actorRef.getSnapshot().status).toBe('error');

    installGlobalOnErrorHandler((ev) => {
      ev.preventDefault();
      expect(ev.error.message).toEqual('unhandled_sync_error_on_state_entry_global');
      done();
    });
  });
```

**Comment by Andarist** [comment, marked as answer]
I think that this open PR resolves both of your concerns: https://github.com/statelyai/xstate/pull/4492 . Could you recheck your expectations with its content?

**Reply by christophe-g** [reply]
Haha - you are brilliant

The above test pass when checked-out from #4492 ! 

Thanks - I'll just wait for the PR to be merged.
